app/components/app-monitoring/settings/index.ts

```diff
@@ -36,6 +76,65 @@ export default class AppMonitoringSettingsComponent extends Component<AppMonitor
       this.appmonitoring.reload();
     }
   });
+
+  get platformObjects() {
+    return [
+      {
+        key: 'All',
+        value: -1,
+      },
+      {
+        key: 'Android',
+        value: ENUMS.PLATFORM.ANDROID,
+      },
+      {
+        key: 'iOS',
+        value: ENUMS.PLATFORM.IOS,
+      },
+    ];
+  }
+
+  get dropDownClass() {
+    return styles['filter-input-dropdown'];
+  }
+
+  get triggerClass() {
+    return styles['filter-input'];
+  }
+
+  @action onSearchQueryChange(event: Event) {
+    const query = (event.target as HTMLSelectElement).value;
+    this.searchQuery = query;
+
+    debounce(this, this.setSearchQuery, query, 500);
+  }
+
+  @action clearSearchInput() {
+    this.searchQuery = '';
+
+    this.setSearchQuery(this.searchQuery);
+  }
+
+  setSearchQuery(query: string) {
+    const searchQueryParam = query ? query : null;
+
+    this.router.transitionTo('authenticated.dashboard.app-monitoring', {
+      queryParams: { app_offset: 0, app_query: searchQueryParam },
+      merge: true,
+    });
+  }
+
+  @action filterPlatform(platform: PlatformObject) {
+    this.selectedPlatform = platform;
+    this.platform = platform.value;
```

Comment:
why is `this.platform` required ? we can anyways use `this.selectedPlatform.value`